### ISSUE TITLE
Video file uploaded to wrong directory

### ISSUE THREAD

**Issue by ftorrisi** [flingo64/PhotoStation-Upload-Lr-Plugin#11]
Hi Flingo64,

First of all thank you for developing this plugin I really find it great.

I have noticed a bug with uploading video files that end-up in the wrong directory. 
I have also a proposed fix. The log captured during the error is attached for your reference.

How to reproduce: I have a few pictures and videos taken on the same day. The target album is Collection/{Date %Y/%m}. The pictures are uploaded in the right folder whereas the videos in the wrong one.

I have been looking a bit at the code and I think the problem is on line 248 of the file PSUtilities.lua:
srcPhotoDate = srcPhoto:getRawMetadata("dateTimeOriginal")

This call could actually be replaced by something a bit more robust along the lines of what is done in the function getDateTimeOriginal.

I am afraid that my camera does fill only "datetimedigitalized" and not "datetimeoriginal" .

If you could address this bug, your tool would be even better!

Thank you
Fabio

[log.txt](https://github.com/flingo64/PhotoStation-Upload-Lr-Plugin/files/85201/log.txt)

**Comment by flingo64** [comment]
Hi Fabio,

you are right, {Date format} could be more robust. 

I didn't think about it, because I already added the correct DateTimeOriginal to all my videos in Lightroom via Metadata --\>Edit Capture Time. This has the advantage, that the videos get the right chronological sorting in Lr.

Anyway, could you please check if this code modification solves the problem:
In PSUtilities.lua, change line 248:
```
			srcPhotoDate = srcPhoto:getRawMetadata("dateTimeOriginal")
```

to:
```
			if srcPhoto:getRawMetadata('isVideo') then
				srcPhotoDate = LrDate.timeFromPosixDate(getDateTimeOriginal(path, srcPhoto))
			else
				srcPhotoDate = srcPhoto:getRawMetadata("dateTimeOriginal")
			end
```
If so, it will be incorporated in the next release

**Comment by ftorrisi** [comment]
Hallo Martin,

Great it works! Thank you for the prompt answer and fix. Really nice piece of software.

Viele Grüsse,
Fabio

**Comment by flingo64** [comment]
Hi Fabio,
thanks for the feedback! I'll include the fix in the next release and keep this issue open just in case anybody else realizes the same problem in the meantime.

Ciao ragazzo!

;-) Martin

**Comment by flingo64** [comment]
The fix mentioned above won't work for photos/videos that don't even have a DateTimeDigitized, where the last possible timestamp is the file creation time. The correct is: 

In PSUtilities.lua, change line 248
```
    srcPhotoDate = srcPhoto:getRawMetadata("dateTimeOriginal")
```
to
```
srcPhotoDate = LrDate.timeFromPosixDate(getDateTimeOriginal(srcPhoto:getRawMetadata("path"), srcPhoto))
```
Anyway, this wil be fixed in the next release.

Martin

**Comment by flingo64** [comment]
The fix is incorporated in Version 4.0.0
